arch/arm/Kconfig

```diff
@@ -6,16 +6,25 @@
 menu "ARM Options"
 	depends on ARM
 
-rsource "core/aarch32/Kconfig"
-rsource "core/aarch64/Kconfig"
-
 config ARCH
 	default "arm"
 
 config ARM64
 	bool
 	select 64BIT
 
+config CPU_CORTEX
+	bool
+	help
+	  This option signifies the use of a CPU of the Cortex family.
+
+config CPU_CORTEX_A
```

Comment:
@stephanosio this is clearly, not ideal, though.
Cortex-A is the only type that is defined in this file, I wonder if you can move this back, inside the aarch64 Kconfig

Reply:
That might be needed, actually, if CPU_CORTEX_A needs different selections depending on whether it is 64 or 32 bit.

Reply:
> this is clearly, not ideal, though.

I understand this can seem that way from the coherency perspective; though, Cortex-A arguably deserves a special treatment because it is the only Cortex family architecture that supports both AArch32 and AArch64 (at the time of writing).

> That might be needed, actually, if CPU_CORTEX_A needs different selections depending on whether it is 64 or 32 bit.

Ideally, `CPU_CORTEX_A` in itself should probably not imply/select execution mode-specific (i.e. AArch32 vs. AArch64) configurations, as it refers to the series of the processor and not its active execution mode; though, we do not currently have a clear picture on how this all would work out (or if this is really necessary at all).

I will move `CPU_CORTEX_A` back to AArch64 for now, as we do not have the full picture at the moment and, as long as that is the case, it can be argued that this is an incoherency.

Reply:
Done